File: projects/10/ruby/Bird.rb

# Bird.rb | Defines a Bird superclass 
#   to be extended by specific bird sub-classes.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Daniela Contreras
# Date:03/25/2022
####################################################

class Bird

    attr_reader :name
  
    def initialize(name)
      @name = name
    end

    def call
        'Squaaaaaaawk!'
    end
    
    def className
        self.class.to_s
    end
    def movement
        "strolled by"
    end
    def print
        puts name + " " + className + "kindly " + movement + " and says " + call
    end
  
end

File: projects/10/java/Bird.java
/* Bird.java provides a Bird class.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Daniela Contreras
 * Date:03/30/22
 ******************************************************/

import java.io.*;

public abstract class Bird
{

  public Bird()
  {
    myName = "";
  } 

  public Bird(String name)
  {
    myName = name;
  } 


  public String getName()
  {
    return myName;
  } 

  public String call()
  {
    return "Squaaaaaaaaaawk!";
  } 

  public abstract String Movement();

  public void print()
  {
    System.out.println( getName() + ' ' + getClass().getName() + ' ' +
                Movement() + " and says " + call() );
  } 


  private String myName;
}

